pcap-common.c

```diff
@@ -1338,7 +1338,7 @@ swap_nflog_header(const struct pcap_pkthdr *hdr, u_char *buf)
 		return;
 	}
 
-	if (nfhdr->nflog_version != 0) {
+	if ((!nfhdr->nflog_version) == 0) {
```

Comment:
The code is checking whether the NFLOG version, which is stored in the `nflog_version` member of the `nflog_hdr_t` structure pointed to by `nfhdr`, is 0, which is the version we support.

The way you check whether the `nflog_version` member of the `nflog_hdr_t` structure pointed to by `nfhdr`, is 0 is by doing `nfhdr->nflog_version != 0`.  If your compiler can't handle that, get a better compiler.